### DISCUSSION TITLE
Creating a custom dashboard widget

### DISCUSSION THREAD

**Discussion by amity2002** [netbox-community/netbox#14210] [Help Wanted!]
Hey everyone,

I want to create a dashboard widget for something external I need. I have followed the full tutorial on the netbox documentation page regarding dashboard registration, yet something isn't working as intended.

Taking the example plugin from the netbox page actually worked, however, it doesn't register the actual dasboard itself (when I press 'Add Widget' I don't see the new Widget Type)

After I do make the widget, do I need to register it somewhere in netbox itself?

setup.py:

```
from setuptools import find_packages, setup

setup(
    name='netbox-access-lists',
    version='0.3',
    description='An example NetBox plugin',
    install_requires=[],
    packages=find_packages(),
    include_package_data=True,
    zip_safe=False,
)
```

netbox_access_lists/dashboard.py:

```
from django import forms
from extras.dashboard.utils import register_widget
from extras.dashboard.widgets import DashboardWidget, WidgetConfigForm


@register_widget
class ReminderWidget(DashboardWidget):
    default_title = 'Reminder'
    description = 'Add a virtual sticky note'

    class ConfigForm(WidgetConfigForm):
        content = forms.CharField(
            widget=forms.Textarea()
        )

    def render(self, request):
        return self.config.get('content')
```

netbox_access_lists/__init__.py:

```
from extras.plugins import PluginConfig

class NetBoxAccessListsConfig(PluginConfig):
    name = 'netbox_access_lists'
    verbose_name = ' NetBox Access Lists'
    description = 'Manage simple ACLs in NetBox'
    version = '0.3'
    base_url = 'access-lists'


config = NetBoxAccessListsConfig
```

**Comment by mulmat** [comment]
I had the same issue, see [this PR](https://github.com/netbox-community/netbox/pull/14913) for solution
